### PR TITLE
Fixed the failing tests due to the outcome template changes and have …

### PR DESCRIPTION
**Pull request by SteveScorfield**
…added resilience tests

### CODE REVIEW

File: src/test/resources/acceptancetests/Update.feature [hunk 1]
```diff
@@ -66,12 +66,12 @@ Feature: SPG Update Tests
...
 
-  Scenario: As Gateway I can receive an Pause Case to a request from RM for an existing Household job
-    Given a TM doesnt have a job with case ID "bd6345af-d706-43d3-a13b-8c549e081a76" in TM
-    And RM sends a HH create job request 
-    And RM sends a HH Pause Case request for the case
-    When Gateway receives an HH Pause Case message for the case
-		And is Processed as "HH Pause Case"
-    Then it will Pause the job in TM
-    And the Paused job is acknowledged by TM
+#  Scenario: As Gateway I can receive an Pause Case to a request from RM for an existing Household job
```
Comment (spdiaz1000) [inline src/test/resources/acceptancetests/Update.feature:69]: Why is this one commented out?

File: src/test/resources/acceptancetests/Outcome.feature [hunk 2]
```diff
@@ -15,83 +15,82 @@ Feature: Outcome Tests
...
+      | SurveyType | BusinessFunction      | Primary Outcome | Secondary Outcome                       | Outcome Code | HasLinkedQID | HasFulfilmentRequest | Operation List                                                                 | RmMessages                                                 | JsMessages |
+    | SPG        | Not Valid Address     | Not Valid       | Visit - Unoccupied Site                 | 6-30-03      | F            | F                    | ADDRESS_NOT_VALID,CANCEL_FEEDBACK,FULFILMENT_REQUESTED,LINKED_QID              | ADDRESS_NOT_VALID                                            | CANCEL     |
+#    | SPG        | Not Valid Address     | Not Valid       | Visit - Unoccupied Site                 | 6-30-03      | T            | F                    | ADDRESS_NOT_VALID,CANCEL_FEEDBACK,FULFILMENT_REQUESTED,LINKED_QID              | ADDRESS_NOT_VALID,QUESTIONNAIRE_LINKED                       | CANCEL     |
```
Comment (spdiaz1000) [inline src/test/resources/acceptancetests/Outcome.feature:20]: all these are commented out

File: src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+  }
+
+  @Then("this message will have the {string} in the message cache and {string} in the gateway cache")
+  public void checkActionInMessageCache(String actionGateway, String actionMessage) throws Exception {
+    String caseId = testBucket.get("caseId");
+    int checkGatewayCache;
+    int checkMessageCache = 0;
+
+    switch (actionGateway) {
+      case "Create":
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java:240]: in theory you could collapse it so it reads
```
  switch (actionGateway) {
      case "Create":      
      case "Update":
      case "Cancel":
        checkGatewayCache = tmMockUtils.checkActionExistsInGatewayCache(actionGateway, caseId);
        break;
```

File: src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+  }
+
+  @Then("the gateway will not process the stored message the message")
+  public void gatewayWillNotProcessMessage() {
+    String caseId = testBucket.get("caseId");
+    boolean hasBeenTriggered = gatewayEventMonitor.hasErrorEventTriggered(caseId, ROUTING_FAILED, CommonUtils.TIMEOUT);
+    assertThat(hasBeenTriggered).isTrue();
+  }
+
+//  @Given("a TM doesnt have a job with case ID {string} in TM")
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java:276]: commented out

**Review by spdiaz1000 — APPROVED** [review]
Well done on the Acceptance Tests, looks like youre thinking a bit clearer in the way the code can be structured.
Theres an issue of having to cuplicate the cache code in 3 places. Something for technical debt

File: src/test/resources/acceptancetests/Update.feature (hunk 1 above)
Comment (SteveScorfield) [reply src/test/resources/acceptancetests/Update.feature:69]: Uncommented.

File: src/test/resources/acceptancetests/Outcome.feature (hunk 2 above)
Comment (SteveScorfield) [reply src/test/resources/acceptancetests/Outcome.feature:20]: Uncommented

File: src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java (hunk 4 above)
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java:276]: Removed this.

File: src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java (hunk 3 above)
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java:240]: I condensed these as suggested.

**Pull request merged**